lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
> all these problems might just melt away :sunglasses:

So... [do they](https://gist.github.com/rhattersley/bed9164759cc827a108c#file-use_cube-py)? This PR was inspired by a user issue wasn't it? How would that shake down with a deconstructed save? And which file format(s) would they need?